### PR TITLE
Make edit boxes respond to string input (IME)

### PR DESCRIPTION
**Pull request by yw05**
This PR makes edit boxes respond to string input events (introduced in minetest/irrlicht#23) that are usually triggered by entering text with an IME.

## To do

This PR is ready for review.

## How to test
- Compile with the `master` branch of minetest/irrlicht
- Try to enter text with an IME

### CODE REVIEW

**Comment by sfan5** [comment]
So the CI doesn't immediately fall over you can put all code that references `EET_STRING_INPUT_EVENT` inside `#if IRRLICHT_VERSION_MT_REVISION > 1`.

**Comment by yw05** [comment]
I noticed that some time ago as well. I am currently testing to make sure it works on my side (just my habit) before pushing it to my fork.

File: src/gui/guiChatConsole.cpp [hunk 1]
```diff
@@ -619,6 +620,13 @@ bool GUIChatConsole::OnEvent(const SEvent& event)
 			m_chat_backend->scroll(rows);
 		}
 	}
+#if defined(IRRLICHT_VERSION_MT_REVISION) && (IRRLICHT_VERSION_MT_REVISION >= 2)
```
Comment (sfan5) [inline src/gui/guiChatConsole.cpp:623]: Since this is client code you can count on `IRRLICHT_VERSION_MT_REVISION` being defined.

Comment (yw05) [reply src/gui/guiChatConsole.cpp:623]: Weirdly, the CI did complain about the missing string input events in the chat console and the edit boxes

**Comment by yw05** [comment]
I am not sure if this is to be expected, but the `android/build` test appears to always fail. I'm not sure how to fix that.

File: src/gui/guiEditBox.cpp [hunk 2]
```diff
@@ -216,6 +217,12 @@ bool GUIEditBox::OnEvent(const SEvent &event)
 			if (processMouse(event))
 				return true;
 			break;
+#if (IRRLICHT_VERSION_MT_REVISION >= 2)
+		case EET_STRING_INPUT_EVENT:
+			inputString(*event.StringInput.Str);
+			return true;
+			break;
```
Comment (sfan5) [inline src/gui/guiEditBox.cpp:224]:
```suggestion
```

File: src/gui/guiEditBox.cpp [hunk 3]
```diff
@@ -670,39 +677,46 @@ bool GUIEditBox::onKeyDelete(const SEvent &event, s32 &mark_begin, s32 &mark_end
...
+	if (!isEnabled() || !m_writable)
+		return;
 
-			m_blink_start_time = porting::getTimeMs();
-			setTextMarkers(0, 0);
+	u32 len = str.size();
+	if (Text.size()+len <= m_max || m_max == 0) {
+		core::stringw s;
+		if (m_mark_begin != m_mark_end) {
+			// clang-format off
```
Comment (sfan5) [inline src/gui/guiEditBox.cpp:695]: (these are no longer needed)
```suggestion
```

File: src/gui/guiEditBox.cpp [hunk 4]
```diff
@@ -670,39 +677,46 @@ bool GUIEditBox::onKeyDelete(const SEvent &event, s32 &mark_begin, s32 &mark_end
...
+			// replace marked test
+			s32 real_begin = m_mark_begin < m_mark_end ? m_mark_begin : m_mark_end;
+			s32 real_end = m_mark_begin < m_mark_end ? m_mark_end : m_mark_begin;
+
+			s = Text.subString(0, real_begin);
+			s.append(str);
+			s.append(Text.subString(real_end, Text.size() - real_end));
+			Text = s;
+			m_cursor_pos = real_begin + len;
+			// clang-format on
```
Comment (sfan5) [inline src/gui/guiEditBox.cpp:705]:
```suggestion
```

**Review by sfan5 — APPROVED** [review]
Works.
(also no the Android CI failure is not your fault)

**Review by SmallJoker — APPROVED** [review]

**Pull request merged**